### PR TITLE
Separate PartitionWriter into interface and Disk implementation

### PR DESCRIPTION
**Pull request by loffek**
- created a `PartitionWriter` interface
- introduced a new PartitionException to avoid leaking implementation details
- moved the old PartitionWriter to `partition.disk` package and named it DiskPartitionWriter

How to test:

- compile the jar: `./gradlew shadowJar`
- replace the jar in https://github.com/getdreams/kafka-backup-wip/tree/master/connect
- follow README: https://github.com/getdreams/kafka-backup-wip/blob/master/README.txt

### CODE REVIEW

File: src/main/java/de/azapps/kafkabackup/common/partition/PartitionException.java [hunk 1]
```diff
@@ -0,0 +1,10 @@
+package de.azapps.kafkabackup.common.partition;
+
+
+public class PartitionException extends Exception {
```
Comment (maciejmiedzianowski) [inline src/main/java/de/azapps/kafkabackup/common/partition/PartitionException.java]: RuntimeException instead?

File: src/main/java/de/azapps/kafkabackup/common/partition/PartitionWriter.java [hunk 2]
```diff
@@ -1,76 +1,14 @@
...
-    }
-    public String topic() {
-        return topic;
-    }
-
-    public int partition() {
-        return partition;
-    }
+public interface PartitionWriter {
+    public void append(Record record) throws PartitionException;
```
Comment (maciejmiedzianowski) [inline src/main/java/de/azapps/kafkabackup/common/partition/PartitionWriter.java]: modifier "public" is redundant for methods in interfaces - each method by default is public

File: src/main/java/de/azapps/kafkabackup/common/partition/PartitionException.java (hunk 1 above)
Comment (loffek) [reply src/main/java/de/azapps/kafkabackup/common/partition/PartitionException.java]: You mean replacing the PartitionException with RuntimeException, or inherit from RuntimeException?

Comment (maciejmiedzianowski) [reply src/main/java/de/azapps/kafkabackup/common/partition/PartitionException.java]: inherit from RuntimeException

**Review by maciejmiedzianowski — APPROVED** [review]

**Pull request merged**